flang/lib/Evaluate/check-expression.cpp

```diff
@@ -188,6 +188,12 @@ struct IsActuallyConstantHelper {
     return common::visit([this](const auto &y) { return (*this)(y); }, x.u);
   }
   bool operator()(const StructureConstructor &x) {
+    // If the comp-spec of a derived-type-spec is a structure constructor that
```

Comment:
"derived-type-spec" is a grammar production associated with types.  I don't know how that can be relevant here.  And I think that `UnwrapExpr<Constant<SomeDerived>>(x)` will always be null -- if the subexpression is a structure constructor, it's not a `Constant` -- so the effect is to just return `false` for all structure constructors.  So I don't understand this part at all.  What are you trying to accomplish here?

Reply:
Thanks for the quick review. I was trying to handle the following case. My original approach is not handling it correctly. I will update the patch with the better approach shortly
```fortran
type dt0
  integer, allocatable :: a(:)
end type
type dt1
  type(dt0) :: b
end type
type(dt1) xdt1 = dt1(dt0(5))
```